mne/io/nirx/nirx.py

```diff
@@ -0,0 +1,204 @@
+            last_sample += 1
+
+        # Read participant information file
+
+        inf = cp.ConfigParser(allow_no_value=True)
+        inf.read(file_inf)
+        inf = inf._sections['Subject Demographics']
+
+        # mne requires specific fields in here
+        # https://github.com/mne-tools/mne-python/ ...
+        # blob/master/mne/io/meas_info.py#L430
+        # TODO: Can you put more values in subject_info than specified in link?
+        #       NIRX also records "Study Type", "Experiment History",
+        #       "Additional Notes", "Contact Information"
+        subject_info = {}
+        subject_info['last_name'] = inf['name'].split()[-1].replace("\"", "")
+        subject_info['first_name'] = inf['name'].split()[0].replace("\"", "")
+        subject_info['middle_name'] = inf['name'].split()[-2].replace("\"", "")
+        subject_info['birthday'] = inf['age']
+        subject_info['sex'] = inf['gender'].replace("\"", "")
+        # Recode values
+        if subject_info['sex'] in {'M', 'Male', '1'}:
+            subject_info['sex'] = 1
+        if subject_info['sex'] in {'F', 'Female', '2'}:
+            subject_info['sex'] = 2
+        # NIRStar does not record an id, or handedness by default
+
+        # Read header file
+        # This is a bit tricky as the header file isnt compliant with
+        # the config specifications. So we need to remove all text
+        # between comments before passing to config parser
+        hdr_str = open(file_hdr[0]).read()
+        hdr_str = re.sub('#.*?#', '', hdr_str, flags=re.DOTALL)
+        hdr = cp.RawConfigParser()
+        hdr.read_string(hdr_str)
+
+        # Check that the file format version is supported
+        assert (hdr['GeneralInfo']['NIRStar'] == "\"15.2\""), \
+            "Only NIRStar version 15.2 is supported"
+
+        # Parse required header fields
+
+        # Extract source-detectors requested by user
+        sources = [int(s) for s in re.findall(r'(\d)-\d:\d+',
+                   hdr['DataStructure']['S-D-Key'])]
+        detectors = [int(s) for s in re.findall(r'\d-(\d):\d+',
+                     hdr['DataStructure']['S-D-Key'])]
+        sdindex = [int(s) for s in re.findall(r'\d-\d:(\d+)',
+                   hdr['DataStructure']['S-D-Key'])]
+        assert (len(sources) == len(detectors)), \
+            "Same amount of sources and detectors required"
+        assert (len(sources) == len(sdindex)), \
+            "Same amount of sources and keys required"
+
+        # Create mne structure
+        # TODO: ch_type is currently misc as I could not find appropriate
+        #       other type, the hbo and hbr type are not relevant until the
+        #       signal has been converted
+        # TODO: nchan needs to be multiplied by two as we have two wavelengths
+        #       per sensor should the underlying type be modified to
+        #       support (wavelength x channels x data)?
+        info = create_info(len(sources) * 2,
```

Comment:
Let's try to think of some sensible names. Maybe `FNIRS_001A, FNIRS_001B, FNIRS_002A, ...`. Even better if we can use something descriptive instead of A and B. S/R for send/receive? What are the two per channel?